File: apps/api/app/models/hook.py
from pydantic import BaseModel
from typing import Optional


class HookItem(BaseModel):
    id: str
    text: str
    language: str
    niche: str
    city: Optional[str] = None
    hook_type: Optional[str] = None
    source: str
    use_count: int
    saturation_score: float
    performance_score: float
    last_used_at: Optional[str] = None
    created_at: str


class HookCreate(BaseModel):
    text: str
    language: str = "tanglish"
    niche: str
    city: Optional[str] = None
    hook_type: Optional[str] = None

File: packages/shared/types/hook.ts
export interface Hook {
  id: string;
  text: string;
  language: string;
  niche: string;
  city: string | null;
  hook_type: string | null;
  source: 'manual' | 'scraped' | 'generated';
  use_count: number;
  saturation_score: number;
  performance_score: number;
  last_used_at: string | null;
  created_at: string;
}

export interface HookCreateInput {
  text: string;
  language?: string;
  niche: string;
  city?: string;
  hook_type?: string;
}

export type HookType = 'curiosity' | 'urgency' | 'local' | 'problem-solution' | 'social-proof';
